src/WinRT.Runtime/IWinRTObject.net5.cs

```diff
@@ -1,188 +1,270 @@
+            }
+
+            if (typeof(System.Collections.IEnumerable).IsAssignableFrom(type))
+            {
+                RuntimeTypeHandle projectIEnum = typeof(System.Collections.IEnumerable).TypeHandle;
+                AdditionalTypeData.GetOrAdd(projectIEnum, (_) => new ABI.System.Collections.IEnumerable.AdaptiveFromAbiHelper(type, this));
+            }
+
+            var vftblType = helperType.FindVftblType();
+            using (objRef)
+            {
+                if (vftblType is null)
+                {
+                    var qiObjRef = objRef.As<IUnknownVftbl>(GuidGenerator.GetIID(helperType));
+                    if (!QueryInterfaceCache.TryAdd(interfaceType, qiObjRef))
+                    {
+                        qiObjRef.Dispose();
+                    }
+                    return true;
+                }
+                IObjectReference typedObjRef = (IObjectReference)typeof(IObjectReference).GetMethod("As", Type.EmptyTypes).MakeGenericMethod(vftblType).Invoke(objRef, null);
+                if (!QueryInterfaceCache.TryAdd(interfaceType, typedObjRef))
+                {
+                    typedObjRef.Dispose();
+                }
+                return true;
+            }
+        }
+
+#if NET8_0_OR_GREATER
+        unsafe bool LookupGeneratedVTableInfo(RuntimeTypeHandle interfaceType, [NotNullWhen(true)] out IIUnknownCacheStrategy.TableInfo? result, out int qiResult)
+        {
+            result = null;
+            qiResult = 0;
+            if (AdditionalTypeData.TryGetValue(interfaceType, out object value))
+            {
+                if (value is IIUnknownCacheStrategy.TableInfo tableInfo)
+                {
+                    result = tableInfo;
+                    return true;
+                }
+                return false;
+            }
+            if (StrategyBasedComWrappers.DefaultIUnknownInterfaceDetailsStrategy.GetIUnknownDerivedDetails(interfaceType) is IIUnknownDerivedDetails details)
+            {
+                qiResult = StrategyBasedComWrappers.DefaultIUnknownStrategy.QueryInterface((void*)NativeObject.ThisPtr, details.Iid, out void* ppv);
+                if (qiResult < 0)
+                {
+                    return false;
+                }
+                var obj = (void***)ppv;
+                result = new IIUnknownCacheStrategy.TableInfo()
+                {
+                    ThisPtr = obj,
+                    Table = *obj,
+                    ManagedType = details.Implementation.TypeHandle
+                };
+
+                if (!AdditionalTypeData.TryAdd(interfaceType, result))
+                {
+                    bool found = AdditionalTypeData.TryGetValue(interfaceType, out object newInfo);
+                    System.Diagnostics.Debug.Assert(found);
+                    result = (IIUnknownCacheStrategy.TableInfo)newInfo;
+                    StrategyBasedComWrappers.DefaultIUnknownStrategy.Release(ppv);
+                }

```

Comment:
Instead of using the `DefaultIUnknownStrategy` here, you should go down the same path as the rest of the CsWinRT interface types to handle the QI if possible (so as to handle COM marshalling, agile objects, etc). The only default strategy you need to use here is the interface details strategy.

Reply:
Should be fixed now, as `IObjectReference.TryAs` is now used.